backend/src/apiserver/resource/resource_manager_test.go

```diff
@@ -70,10 +71,27 @@ func (m *FakeBadObjectStore) GetFromYamlFile(o interface{}, filePath string) err
 var testWorkflow = util.NewWorkflow(&v1alpha1.Workflow{
 	TypeMeta:   v1.TypeMeta{APIVersion: "argoproj.io/v1alpha1", Kind: "Workflow"},
 	ObjectMeta: v1.ObjectMeta{Name: "workflow-name", UID: "workflow1", Namespace: "ns1"},
-	Spec:       v1alpha1.WorkflowSpec{Arguments: v1alpha1.Arguments{Parameters: []v1alpha1.Parameter{{Name: "param1"}}}},
-	Status:     v1alpha1.WorkflowStatus{Phase: v1alpha1.NodeRunning},
+	Spec: v1alpha1.WorkflowSpec{
+		Entrypoint: "testy",
+		Templates: []v1alpha1.Template{v1alpha1.Template{
+			Name: "testy",
+			Container: &corev1.Container{
+				Image:   "docker/whalesay",
+				Command: []string{"cowsay"},
+				Args:    []string{"hello world"},
+			},
+		}},
+		Arguments: v1alpha1.Arguments{Parameters: []v1alpha1.Parameter{{Name: "param1"}}}},
+	Status: v1alpha1.WorkflowStatus{Phase: v1alpha1.NodeRunning},
 })
 
+func setMetadata(wf *v1alpha1.Workflow) {
```

Comment:
nit: addRuntimeWorkflowMetadata? to make tests easier to understand

Reply:
Probably we can omit workflow, addRuntimeMetadata? WDYT?

Reply:
We can set it directly on the test workflow and drop the method to clean up.

Reply:
Hi @NikeNano, note the current change is against the goal we want to test. In real world, the workflows passed from KFP compiler do not have these metadata annotations. Therefore, always adding them beforehand makes tests deviate from the real world.

I understand we all want tests to be simpler, so they can be more readable. I think a better way to do this is to stop comparing all fields of a workflow in tests. Instead, we can reset fields we do not care about in some tests to empty before comparing with expected workflow. e.g. for this case, we can simply set annotations and labels to empty, and do not include them in the expected workflow.

Note that, for tests that specifically verify labels & annotations are correct, we still keep them to test what we want.

Reply:
> Therefore, always adding them beforehand makes tests deviate from the real world.

Yes good point, will have to find another solution. 

I see your point, the tricky part here is that the `WorkflowRuntimeManifest` is handled as a string. Which means we need to unmarshal it to a `*v1alpha1.Workflow` again, remove the metadata and convert to a string again and then set it on the `runDetails`. I think this adds more complexity than just adding it atm.